packages/wordcount/src/index.js

```diff
@@ -17,18 +17,29 @@ import stripShortcodes from './stripShortcodes';
 import stripSpaces from './stripSpaces';
 import transposeHTMLEntitiesToCountableChars from './transposeHTMLEntitiesToCountableChars';
 
+/**
+ * @typedef {import('./defaultSettings').WPWordCountDefaultSettings}  WPWordCountSettings
+ * @typedef {import('./defaultSettings').WPWordCountUserSettings}     WPWordCountUserSettings
+ */
+
+/**
+ * Possible ways of counting.
+ *
+ * @typedef {'words'|'characters_excluding_spaces'|'characters_including_spaces'} WPWordCountStrategy
+ */
+
 /**
  * Private function to manage the settings.
  *
- * @param {string} type         The type of count to be done.
- * @param {Object} userSettings Custom settings for the count.
+ * @param {WPWordCountStrategy}     type         The type of count to be done.
+ * @param {WPWordCountUserSettings} userSettings Custom settings for the count.
  *
- * @return {void|Object|*} The combined settings object to be used.
+ * @return {WPWordCountSettings} The combined settings object to be used.
  */
 function loadSettings( type, userSettings ) {
 	const settings = extend( defaultSettings, userSettings );
```

Comment:
This looks like a (pre-existing) bug, `extend` is an alias for [`assignIn`](https://lodash.com/docs/4.17.15#assignIn):

> This method is like _.assign except that it iterates over own and inherited source properties.
> 
> Note: This method mutates object.

We shouldn't be mutating shared default settings 😱 

```suggestion
	const settings = extend( {}, defaultSettings, userSettings );
```

That's grounds for it's own PR, but since we're deep into a significant refactor, we might sneak it in 🙂